ssl/s3_cbc.c

```diff
@@ -227,15 +228,15 @@ int ssl3_cbc_digest_record(const EVP_MD_CTX *ctx,
          * ssl3_cbc_record_digest_supported should have been called first to
          * check that the hash function is supported.
          */
-        OPENSSL_assert(0);
-        if (md_out_size)
+        if (md_out_size != NULL)
             *md_out_size = 0;
-        return 0;
+        return ossl_assert(0);
     }
 
-    OPENSSL_assert(md_length_size <= MAX_HASH_BIT_COUNT_BYTES);
-    OPENSSL_assert(md_block_size <= MAX_HASH_BLOCK_SIZE);
-    OPENSSL_assert(md_size <= EVP_MAX_MD_SIZE);
+    if (!ossl_assert(md_length_size <= MAX_HASH_BIT_COUNT_BYTES
+                    && md_block_size <= MAX_HASH_BLOCK_SIZE
+                    && md_size <= EVP_MAX_MD_SIZE))
+        return 0;
 
```

Comment:
Is it helpful to know which condition failed?  To keep any assertion strings concise?
Perhaps better:
```
if (!ossl_assert(cond1)
    || !ossl_assert(cond2)
    || ... )
```